alibi_detect/cd/pytorch/classifier.py

```diff
@@ -149,9 +145,24 @@ def score(self, x: np.ndarray) -> float:
             train_args = [model, nn.CrossEntropyLoss(), dl_tr, self.device]
             trainer(*train_args, **self.train_kwargs)  # type: ignore
             preds = predict_batch(x_te, model.eval(), device=self.device, batch_size=self.dl_kwargs['batch_size'])
-            preds_oof.append(preds)
-            idx_oof.append(idx_te)
-        preds_oof = np.concatenate(preds_oof, axis=0).argmax(axis=1)
-        idx_oof = np.concatenate(idx_oof, axis=0)
-        drift_metric = self.metric_fn(y[idx_oof], preds_oof)
-        return drift_metric
+            preds_oof_list.append(preds)
+            idx_oof_list.append(idx_te)
+        preds_oof = np.concatenate(preds_oof_list, axis=0).argmax(axis=1)
+        idx_oof = np.concatenate(idx_oof_list, axis=0)
+
+        if self.soft_preds:
+            log_losses_ref = preds_oof[y[idx_oof] == 0]
+            log_losses_cur = preds_oof[y[idx_oof] == 1]
+            dist, p_val = ks_2samp(log_losses_ref, log_losses_cur, alternative='greater')
+        else:
+            baseline_accuracy = max(x_ref.shape[0], x.shape[0]) / (x_ref.shape[0] + x.shape[0])  # exp under null
+            n_oof = idx_oof.shape[0]
+            n_correct = (y[idx_oof] == preds_oof.round()).sum()
```

Comment:
Don't think this needs to be rounded as we already take the argmax earlier?

Reply:
I don't think we do?

Reply:
The mistake is in taking the argmax earlier. I'd removed it for tf but not pt. Thanks for catching.

Reply:
Don't you need that argmax for the soft_preds?